openmdao/components/interp_util/interp_slinear.py

```diff
@@ -216,6 +220,507 @@ def interpolate(self, x):
+        idx : int
+            List of interval indices for x.
+        dtype : object
+            The dtype for vector allocation; used for complex step.
+
+        Returns
+        -------
+        ndarray
+            Interpolation coefficients.
+        """
+        grid = self.grid
+        values = self.values
+        a = np.empty(4, dtype=dtype)
+
+        i_x, i_y = idx
+
+        # Extrapolation
+        n = len(grid[0])
+        if i_x == n - 1:
+            i_x = n - 2
+        elif i_x == -1:
+            i_x = 0
+
+        n = len(grid[1])
+        if i_y == n - 1:
+            i_y = n - 2
+        elif i_y == -1:
+            i_y = 0
+
+        x0 = grid[0][i_x]
+        x1 = grid[0][i_x + 1]
+        y0 = grid[1][i_y]
+        y1 = grid[1][i_y + 1]
+
+        c00 = values[i_x, i_y]
+        c01 = values[i_x, i_y + 1]
+        c10 = values[i_x + 1, i_y]
+        c11 = values[i_x + 1, i_y + 1]
+
+        a[0] = c00 * x1 * y1 - \
+            c01 * x1 * y0 - \
+            c10 * x0 * y1 + \
+            c11 * x0 * y0
+
+        a[1] = (c10 - c00) * y1 + (c01 - c11) * y0
+
+        a[2] = (c01 - c00) * x1 + (c10 - c11) * x0
+
+        a[3] = c01 + c10 - c11 - c00
+
+        rec_vol = 1.0 / ((x0 - x1) * (y0 - y1))
+        return a * rec_vol
+
+    def interpolate_vectorized(self, x_vec, idx):
+        """
+        Compute the interpolated value.
+
+        Parameters
+        ----------
+        x_vec : ndarray
+            The coordinates to interpolate on this grid.
+        idx : int
+            List of interval indices for x.
+
+        Returns
+        -------
+        ndarray
+            Interpolated values.
+        ndarray
+            Derivative of interpolated values with respect to independents.
+        ndarray
+            Derivative of interpolated values with respect to values.
+        ndarray
+            Derivative of interpolated values with respect to grid.
+        """
+        x = x_vec[:, 0]
+        y = x_vec[:, 1]
+        grid = self.grid
+        vec_size = len(x)
+        i_x, i_y = idx
+
+        # Complex Step
+        if self.values.dtype == complex:
+            dtype = self.values.dtype
+        else:
+            dtype = x.dtype
+
+        # extrapolate low
+        i_x[i_x == -1] = 0
+        i_y[i_y == -1] = 0
+
+        # extrapolate high
+        nx, ny = self.values.shape
+        i_x[i_x == nx - 1] = nx - 2
+        i_y[i_y == ny - 1] = ny - 2
+
+        if self.vec_coeff is None:
+            self.coeffs = set()
+            self.vec_coeff = np.empty((nx, ny, 4), dtype=dtype)
+
+        needed = set([item for item in zip(i_x, i_y)])
+        uncached = needed.difference(self.coeffs)
+        if len(uncached) > 0:
+            unc = np.array(list(uncached))
+            uncached_idx = (unc[:, 0], unc[:, 1])
+            a = self.compute_coeffs_vectorized(uncached_idx, dtype)
+            self.vec_coeff[unc[:, 0], unc[:, 1], ...] = a
+            self.coeffs = self.coeffs.union(uncached)
```

Comment:
union creates a new set, so maybe just use `self.coeffs.update(uncached)`

Reply:
I've changed this over in all spots.